api/batch/controller.go

```diff
@@ -320,6 +321,27 @@ func (c *controller) onUpdate(old, new interface{}) {
 	}
 }
 
+func (c *controller) getMLPSecrets(ctx context.Context, predictionJob *models.PredictionJob, namespace string) (map[string]string, error) {
+	secretMap := make(map[string]string)
+	// Retrieve Google Service Account secret from MLP
+	googleServiceAccountSecret, err := c.mlpAPIClient.GetSecretByName(ctx, predictionJob.Config.ServiceAccountName, int32(predictionJob.ProjectID))
```

Comment:
Q: Is there always a serviceaccount attached to the predictionJob config by default?

Reply:
Yep. Hmm, I wouldn't say by default, but rather, requests to the API server to trigger prediction jobs are expected to have a service account name specified. The API server doesn't have any validation checks for that field to be present (unfortunately) but the UI and SDK do. That field always gets accessed here https://github.com/caraml-dev/merlin/blob/967788eb98a0120dd20bec692ab04bcea8f08fc9/api/batch/controller.go#L161 for all requests to trigger prediction jobs.

Reply:
got it!